UpstreamDownstreamDialog.py

```diff
@@ -82,3 +129,10 @@ def __addLayer(self, url, title):
         layer = QgsRasterLayer("url=%s&crs=EPSG:2193&format=image/png"
                                % (url % hydroid), "%s %s" % (title, hydroid), "wms")
         QgsMapLayerRegistry.instance().addMapLayer(layer)
+    
+    def __onProjectRead(self):
+            try:
+                reg = QgsMapLayerRegistry.instance()
+                self.recLayer = reg.mapLayer(QgsProject.instance().readEntry("fishdb", "recLayer", "")[0])
+            except:
+                QMessageBox.warning(self,  "Read Error",  "Could not read specified Entry.")
```

Comment:
AFAICS `self.recLayer` is potentially used in  `__addLayer` regardless of whether the layer was correctly restored from the settings or not. If this is so, please add some error handling, i.e. an informative error message should be shown to the user and the upstream/downstream buttons should not be clickable. Also, this entire code body could be moved directly to the `__init__` function, it does not help much code-cleanness wise to keep it in a separate function.